src/shared/seccomp-util.c

```diff
@@ -1795,3 +1799,62 @@ int seccomp_protect_hostname(void) {
 
         return 0;
 }
+
+int seccomp_restrict_suid_sgid(void) {
+        uint32_t arch;
+        int r;
+
+        SECCOMP_FOREACH_LOCAL_ARCH(arch) {
+                _cleanup_(seccomp_releasep) scmp_filter_ctx seccomp = NULL;
+
+                r = seccomp_init_for_arch(&seccomp, arch, SCMP_ACT_ALLOW);
+                if (r < 0)
+                        return r;
+
+                /* chmod, fchmod, fchmodat */
+
+                for (unsigned bit = 0; bit < 2; bit ++) {
+                        /* Block S_ISUID in the first iteration, S_ISGID in the second */
+                        mode_t m = bit == 0 ? S_ISUID : S_ISGID;
+
+                        r = seccomp_rule_add_exact(
```

Comment:
There are also open(), openat(), creat().

Reply:
you cannot create suid/sgid files with them

Reply:
```C
#define _GNU_SOURCE
#include <unistd.h>
#include <sys/syscall.h>
#include <sys/types.h>
#include <sys/stat.h>
#include <fcntl.h>

int main(int argc, char **argv) {
	syscall(SYS_open, "/tmp/mysuid", O_CREAT, S_IRWXU | S_IRWXG | S_IRWXO | S_ISUID);
	return 0;
}
```
This seems to create the suid/sgid file.

Reply:
hmm, true, but it has to remain empty, since on Linux the suid/sgid bits are dropped on write, iirc.